### PR TITLE
Add border check to crypto/bio/bss_bio.c/bio_write

### PR DESCRIPTION
**Pull request by PeiweiHu**
Add border check num_ < 0 to avoid potential bugs. 

CLA:trivial

##### Checklist

- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

File: crypto/bio/bss_bio.c [hunk 1]
```diff
@@ -267,15 +267,15 @@ static ossl_ssize_t bio_nread(BIO *bio, char **buf, size_t num_)
 
 static int bio_write(BIO *bio, const char *buf, int num_)
 {
+    if (!bio->init || buf == NULL || num_ <= 0)
```
Comment (t8m) [inline crypto/bio/bss_bio.c:270]: This cannot be moved here - we do not allow code before declarations.

Comment (PeiweiHu) [reply crypto/bio/bss_bio.c:270]: If you think it’s a reasonable change, I will change the code to satisfy the requirement. :-)

Comment (t8m) [reply crypto/bio/bss_bio.c:270]: I would say changing `num == 0` to `num_ <= 0` at the original location would be just fine.

Comment (PeiweiHu) [reply crypto/bio/bss_bio.c]: Totally agree. Already changed like you said.

**Review by t8m — APPROVED** [review]

**Comment by t8m** [comment]
I am OK with CLA: trivial.

**Review by kaduk — APPROVED** [review]
Ok as trivial

**Review by paulidale — APPROVED** [review]

**Comment by paulidale** [comment]
Okay with trivial too.

**Comment by t8m** [comment]
Merged to master with squashing commits and adjusting the commit message. Thank you.

**Pull request closed without merging**